### PR TITLE
Feature/query builder request page

### PR DESCRIPTION
**Pull request by OButlerOcc**
added the query builder links to the request page

### CODE REVIEW

**Comment by NigelPalmer** [comment]
- [X] Cohort Query field displayed above fields.
- [X] Data Query field displayed above fields.
- [X] "Create Cohort Query" displayed when there is no query set.
- [ ] "Create Cohort Query" displayed when there is a query with a single rule. See code review. **Please fix.**
- [X] "Create Cohort Query" displayed when there is no query set.
- [ ] "Create Cohort Query" displayed when there is a query with a single rule. See code review. **Please fix.**
- [ ] Clicking on "Create Data Query" after creating the "Cohort Query" loaded the "Cohort Query". This should be blank. **Please fix.**
- [ ] Create query buttons disappeared completely when I had a Cohort and Data query with a single rule in each one. Probably down to using 1 for length checks. **Please fix.**
- [ ] "Back to Request" link goes to the list of requests. This should return the user to the selected request. **Please Fix.**
- [X] "Edit Cohort Query" displayed when there is a cohort query with more than 1 rule.
- [X] "Edit Data Query" displayed when there is a data query with more than 1 rule.
- [X] MEQL displayed correctly when present.

File: src/app/pages/my-request-detail/my-request-detail.component.html [hunk 1]
```diff
@@ -59,6 +61,51 @@ <h1>{{ request.label }}</h1>
...
+
+        <div class="mdm-my-request__query--description">
+          <div *ngIf="dataQuery.rules.length < 1" class="mdm-my-request__query-missing">
+            <p>No Data Query has been created for this request</p>
+          </div>
+
+          <a *ngIf="dataQuery.rules.length < 1" [routerLink]="['queries/', dataQueryType]"
+            ><button mat-raised-button>Create Data Query</button></a
+          >
+          <a *ngIf="dataQuery.rules.length > 1" [routerLink]="['queries/', dataQueryType]"
```
[the remark this answers is not in the document]
Comment (OButlerOcc) [reply src/app/pages/my-request-detail/my-request-detail.component.html]: Switched to >0 and ==0 to be clearer, have done the same for all of them

File: src/app/pages/my-request-detail/my-request-detail.component.html [hunk 2]
```diff
@@ -59,6 +61,56 @@ <h1>{{ request.label }}</h1>
         <p>{{ request.description }}</p>
       </div>
     </div>
+
+    <div class="mdm-my-request__query">
+      <div class="mdm-my-request__query--row row">
+        <div *ngIf="cohortQuery.rules.length > 0" class="mdm-my-request__query-cohort">
+          <mdm-meql-output [query]="cohortQuery">{{
```
Comment (pjmonks) [inline src/app/pages/my-request-detail/my-request-detail.component.html:68]: Interpolating the `cohortQuery | meql` expression as a child of this component is unecessary, the `mdm-meql-output` component handles all of that when bound to the `query` input.

File: src/app/data-explorer/data-explorer.module.ts [hunk 3]
```diff
@@ -77,6 +77,7 @@ import { MeqlOutputComponent } from './meql-output/meql-output.component';
     SearchFiltersComponent,
     CatalogueSearchFormComponent,
     QueryBuilderComponent,
+    MeqlPipe,
```
Comment (pjmonks) [inline src/app/data-explorer/data-explorer.module.ts:80]: Shouldn't need to import the `MeqlPipe`, the `MeqlOutputComponent` should be enough.

File: src/app/pages/my-request-detail/my-request-detail.component.ts [hunk 4]
```diff
@@ -122,10 +133,39 @@ export class MyRequestDetailComponent implements OnInit {
...
   }
 
+  private initialiseRequestQueries() {
+    if (!this.request?.id) {
+      return;
+    }
+
+    this.dataRequests
+      .getQuery(this.request.id, this.cohortQueryType)
+      .pipe()
```
Comment (pjmonks) [inline src/app/pages/my-request-detail/my-request-detail.component.ts:148]: If there are no other RxJs operators to pipe, I think this empty `pipe()` call should be removed.

**Review by pjmonks — CHANGES_REQUESTED** [review]
Spotted some other code issues to rectify please.

**Comment by NigelPalmer** [comment]
I've noticed another issue. If both queries are blank then both buttons say "Create xxxx query". But if one is populated and one is blank then BOTH buttons say "Edit xxxx query" . Only the button next to the populated query should say "Edit xxxx button". The button next to the blank query should say "Create xxxx button".

File: src/app/pages/my-request-detail/my-request-detail.component.html [hunk 5]
```diff
@@ -59,6 +61,51 @@ <h1>{{ request.label }}</h1>
...
+            cohortQuery | meql
+          }}</mdm-meql-output>
+        </div>
+        <div class="mdm-my-request__query--description">
+          <div *ngIf="cohortQuery.rules.length < 1" class="mdm-my-request__query-missing">
```
[the remark this answers is not in the document]
Comment (OButlerOcc) [reply src/app/pages/my-request-detail/my-request-detail.component.html]: fixed

File: src/app/pages/my-request-detail/my-request-detail.component.html [hunk 6]
```diff
@@ -59,6 +61,51 @@ <h1>{{ request.label }}</h1>
...
+            <p>No Cohort Query has been created for this request</p>
+          </div>
+          <a
+            *ngIf="dataQuery.rules.length < 1"
+            [routerLink]="['queries/', cohortQueryType]"
+            ><button mat-raised-button>Create Cohort Query</button></a
+          >
+          <a
+            *ngIf="dataQuery.rules.length > 1"
```
[the remark this answers is not in the document]
Comment (OButlerOcc) [reply src/app/pages/my-request-detail/my-request-detail.component.html]: fixed

File: src/app/pages/my-request-detail/my-request-detail.component.html [hunk 7]
```diff
@@ -59,6 +61,51 @@ <h1>{{ request.label }}</h1>
...
+        </div>
+      </div>
+
+      <div class="mdm-my-request__query--row row">
+        <div *ngIf="dataQuery.rules.length > 0" class="mdm-my-request__query-data">
+          <mdm-meql-output [query]="dataQuery">{{ dataQuery | meql }}</mdm-meql-output>
+        </div>
+
+        <div class="mdm-my-request__query--description">
+          <div *ngIf="dataQuery.rules.length < 1" class="mdm-my-request__query-missing">
```
[the remark this answers is not in the document]
Comment (OButlerOcc) [reply src/app/pages/my-request-detail/my-request-detail.component.html]: fixed

File: src/app/pages/my-request-detail/my-request-detail.component.ts (hunk 4 above)
Comment (OButlerOcc) [reply src/app/pages/my-request-detail/my-request-detail.component.ts:144]: fixed

File: src/app/pages/my-request-detail/my-request-detail.component.html (hunk 2 above)
Comment (OButlerOcc) [reply src/app/pages/my-request-detail/my-request-detail.component.html:73]: I see, fixed

File: src/app/data-explorer/data-explorer.module.ts (hunk 3 above)
Comment (OButlerOcc) [reply src/app/data-explorer/data-explorer.module.ts:80]: Removed as not needed

**Review by NigelPalmer — APPROVED** [review]
@OButlerOcc has double checked my commit and is happy.

**Review by pjmonks — APPROVED** [review]

**Pull request merged**